### PR TITLE
Fix licenses in HTML output

### LINKED ISSUE #3272: Docker run No license be detected！
### Description
I had used https://github.com/nexB/scancode-toolkit/blob/develop/Dockerfile to build docker images .then run it but no any licenses be detected!
NO any licenses information in result.html!

### How To Reproduce

download https://github.com/nexB/scancode-toolkit/archive/refs/tags/v32.0.0rc2.tar.gz
tar zxvf scancode-toolkit-32.0.0rc2.tar.gz
cd scancode-toolkit-32.0.0rc2

docker build --tag scancode-toolkit:latest .
cd samples
docker run --rm -v $PWD/:/project scancode-toolkit:latest -n 10 -clpeui  --html /project/result.html  /project

### System configuration

* What OS are you running on? (Windows/MacOS/Linux)  centos 7.0
* What version of scancode-toolkit was used to generate the scan file? 32.0.0rc2
* What installation method was used to install/run scancode? (pip/source download/other) source download

### PR DESCRIPTION
**Pull request by AyanSinhaMahapatra**
Fixes #3272 

### Tasks

* [x] Reviewed [contribution guidelines](https://github.com/nexB/scancode-toolkit/blob/develop/CONTRIBUTING.rst)
* [x] PR is descriptively titled 📑 and links the original issue above 🔗
* [x] Tests pass -- look for a green checkbox ✔️ a few minutes after opening your PR
  Run [tests](https://scancode-toolkit.readthedocs.io/en/latest/contribute/contrib_dev.html#running-tests) locally to check for errors. 
* [x] Commits are in uniquely-named feature branch and has no merge conflicts 📁

### CODE REVIEW

**Review by pombredanne — CHANGES_REQUESTED** [review]
LGTM, but we need a test

**Comment by jbohrmann** [comment]
Hi, license text is not included in HTML-output. As I am new to scancode I am not sure if this is expected behavior.
I run scancode under windows (py 3.9).
I call it this way (--license-text-diagnostics also does not change the output):
`scancode.bat -cleui -n 2 --html output.html . --license-score 2 --license-text --license-text-diagnostics`

**Comment by AyanSinhaMahapatra** [comment]
@jbohrmann can you confirm that you used the checkout of this branch?

```
git clone https://github.com/nexB/scancode-toolkit.git
git checkout fix-html-output
./configure --dev
 scancode -clipeu --html output.html samples/
```

I seem to have licenses fine from this branch, attaching here the output.html file I generated using these commands above:
[output.html](https://github.com/nexB/scancode-toolkit/files/10850279/output.html.txt)

Also `--html` is a very basic output format, and does not have support for most of the advanced CLI options such as the one you have used:
* --license-score
*  --license-text 
* --license-text-diagnostics
Use --json or --yaml to use all the supported CLI options.

**Comment by jbohrmann** [comment]
> @jbohrmann can you confirm that you used the checkout of this branch?
> 
> ```
> git clone https://github.com/nexB/scancode-toolkit.git
> git checkout fix-html-output
> ./configure --dev
>  scancode -clipeu --html output.html samples/
> ```
> 
> I seem to have licenses fine from this branch, attaching here the output.html file I generated using these commands above: [output.html](https://github.com/nexB/scancode-toolkit/files/10850279/output.html.txt)
> 
> Also `--html` is a very basic output format, and does not have support for most of the advanced CLI options such as the one you have used:
> 
> * --license-score
> * --license-text
> * --license-text-diagnostics
>   Use --json or --yaml to use all the supported CLI options.

Yes I can confirm.
Here is the info from my .VERSION file:

`refs=refs/pull/3275/head, fix-html-output`

I preferred HTML as I can pass the output easily to get legal advice.

**Comment by AyanSinhaMahapatra** [comment]
@jbohrmann what is your OS? also what is your installation process? Is it a git checkout or an app archive download? the .VERSION file looks like this https://github.com/nexB/scancode-toolkit/blob/develop/.VERSION when you are installing from a git checkout, what do you mean info from this file?
Can you do a `git describe --tag` for me and confirm that the output is `v32.0.0rc2-2-g76795acd45` which is the last commit here? 

See above the commit with added test for licenses in html, looks like they work fine now after these changes ^

**Comment by pombredanne** [comment]
@AyanSinhaMahapatra @jbohrmann said it was Windows and Python 3.9 ;)
>  I run scancode under windows (py 3.9).

**Comment by pombredanne** [comment]
but afaics, we do not include the license text be it the matched text or the reference text in the standard HTML output. Maybe we should. HTML is nicer than JSON for sure.

**Comment by AyanSinhaMahapatra** [comment]
@pombredanne We could have support for license-text, summary, license-clarity-score and others too in the HTML output, I opened an issue for this: https://github.com/nexB/scancode-toolkit/issues/3281.

**Comment by jbohrmann** [comment]
I downloaded the branch as zip over Github. OS is Windows 10. 
Maybe it would be helpful to mention in the documentation that at the moment _--html_ does not support all features.

**Comment by pombredanne** [comment]
>  Maybe it would be helpful to mention in the documentation that at the moment --html does not support all features.

@jbohrmann you are right! Note that there are a few alternatives:
- scancode.io provides a web ui to display the matched license text
- it is possible otherwise to create a custom HTML template that would include the matched text

File: src/formattedcode/templates/html/template.html [hunk 1]
```diff
@@ -276,15 +276,19 @@
...
       <tbody>
-        {% for key, license in licenses.items() %}
+        {% for license in licenses %}
           <tr id="license_{{ license.key }}">
             <td>{{ license.key }}</td>
             <td>{{ license.short_name }}</td>
             <td>{{ license.category }}</td>
             <td>{{ license.owner }}</td>
-            <td>{{ license.reference_url|urlize(target='_blank') }}</td>
-            <td>{{ license.homepage_url|urlize(target='_blank') }}</td>
```
Comment (pombredanne) [inline src/formattedcode/templates/html/template.html:286]: May be we should keep the homepage_url

Comment (AyanSinhaMahapatra) [reply src/formattedcode/templates/html/template.html:286]: Added this.

**Review by JonoYang — APPROVED** [review]
LGTM. I've tested out the HTML output on a scan of the samples directory and the results look great. Thanks!

**Comment by AyanSinhaMahapatra** [comment]
Thanks @JonoYang , merging!

**Pull request merged**